synth.py

```diff
@@ -2,84 +2,68 @@
+# surgery in client.ts file to call IAM service
+clients = ['publisher', 'subscriber']
+for client_name in clients:
+    client_file = f'src/v1/{client_name}_client.ts'
+    s.replace(client_file,
+              f'import \* as gapicConfig from \'\.\/{client_name}_client_config\.json\';',
+              f'import * as gapicConfig from \'./%s_client_config.json\';\nimport {{IamClient}} from \'../helper\';' % client_name,
+              )
 
-# The JavaScript generator didn't implement close(). TypeScript gapic does,
-# so this should be removed when we merge that.
-s.replace("src/v1/publisher_client.js",
-          "   \* Parse the projectName from a project resource.\n",
-          "   * Terminate the GRPC channel and close the client.\n"
-          "   * note: editing generated code\n"
-          "   *\n"
-          "   * The client will no longer be usable and all future behavior is undefined.\n"
-          "   */\n"
-          "  close() {\n"
-          "    return this.publisherStub.then(stub => {\n"
-          "      stub.close();\n"
-          "    });\n"
-          "  }\n"
-          "  \n"
-          "  /**\n"
-          "  \g<0>")
-s.replace("src/v1/publisher_client.js",
-          "    const publisherStubMethods = \\[\n",
-          "    // note: editing generated code\n"
-          "    this.publisherStub = publisherStub;\n"
-          "    \g<0>")
+    s.replace(client_file,
+              'private \_terminated = false;',
+              'private _terminated = false; \n private _iamClient: IamClient;')
 
-# Update path discovery due to build/ dir and TypeScript conversion.
-s.replace("src/v1/publisher_client.js", "../../package.json", "../../../package.json")
-s.replace("src/v1/subscriber_client.js", "../../package.json", "../../../package.json")
+    s.replace(client_file,
+              '\/\/ Determine the client header string.',
+              'this._iamClient = new IamClient(opts); \n // Determine the client header string.')
 
-# [START fix-dead-link]
-s.replace('src/**/doc/google/protobuf/doc_timestamp.js',
-        'https:\/\/cloud\.google\.com[\s\*]*http:\/\/(.*)[\s\*]*\)',
-        r"https://\1)")
+    # TODO: it should be removed once pubsub upgrade gts 2.0.0
+    # fix tslint issue due to mismatch gts version with gapic-generator-typescript
+    s.replace(client_file, '\/\/ eslint\-disable\-next\-line\ \@typescript\-eslint\/no\-explicit\-any',
+              '// tslint:disable-next-line no-any')
 
-s.replace('src/**/doc/google/protobuf/doc_timestamp.js',
-        'toISOString\]',
-        'toISOString)')
-# [END fix-dead-link]
+    with open('helperMethods.ts.tmpl', 'r') as helper_file:
+        content = helper_file.read()
+    s.replace(client_file, '^}', content)
 
-# No browser support for TypeScript libraries yet
-os.unlink('webpack.config.js')
-os.unlink('src/browser.js')
+# TODO: Remove this replace once https://github.com/googleapis/gapic-generator-typescript/issues/380 resolved
```

Comment:
It looks like this issue is resolved.  Can we remove the synth change?

Reply:
yes the docker image contains this change now, can you please regen the client? @summer-ji-eng  Thank you!

Reply:
sorry. this is another tslinter error in gts 1.0.0, it could be fix once pubsub upgrade to 2.0.0

Reply:
but I update the comment